adserver/forms.py

```diff
@@ -31,13 +31,40 @@
 from .models import Advertisement
 from .models import Flight
 from .models import Publisher
+from .regiontopics import africa
+from .regiontopics import backend_web
+from .regiontopics import data_science
+from .regiontopics import devops
+from .regiontopics import eu_aus_nz
+from .regiontopics import exclude
+from .regiontopics import frontend_web
+from .regiontopics import latin_america
+from .regiontopics import python
+from .regiontopics import security_privacy
+from .regiontopics import us_ca
+from .regiontopics import wider_apac
+from .validators import TargetingParametersValidator
 
 
 log = logging.getLogger(__name__)  # noqa
 User = get_user_model()
 
 
-class FlightAdminForm(forms.ModelForm):
+class FlightMixin:
+
+    """Ensure the flight can't have both CPC and CPM."""
+
+    def clean(self):
+        cleaned_data = super().clean()
+        cpc = cleaned_data.get("cpc")
+        cpm = cleaned_data.get("cpm")
+        if cpc > 0 and cpm > 0:
```

Comment:
I've wondered about putting the value of the buy into this instead of the number of impressions. I feel like whenever I have to setup a flight, I work backwards from the value, but maybe thats ok. Seems like that's the real number people care about tho -- just a random thought I had the other day.

Reply:
I actually think that is probably the right way.